### PR TITLE
Publish only tested charm

### PR DESCRIPTION
**Pull request by ghislainbourgeois**
# Description

We had an uncaught bug caused by repacking the charm before publishing it to Charmhub. This PR should prevent recurrence by publishing the charm that was built during the integration test job, after those tests pass. This will insure that we publish the exact charm we tested.

# Checklist:

- [x] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that validate the behaviour of the software
- [x] I validated that new and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -50,3 +50,27 @@ jobs:
...
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Fetch Tested Charm
+        uses: actions/download-artifact@v3
+        with:
+          name: tested-charm
+      - name: Upload charm to Charmhub edge channel
+        env:
+          CHARMCRAFT_AUTH: "${{ secrets.CHARMHUB_TOKEN }}"
+        run: charmcraft upload ./tls-certificates-operator_ubuntu-20.04-amd64.charm --release edge
```
Comment (saltiyazan) [inline .github/workflows/main.yml:76]: Wouldn't this on pull request creation?

**Review by saltiyazan — APPROVED** [review]

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -50,3 +50,27 @@ jobs:
...
+    runs-on: ubuntu-20.04
+    if: ${{ github.event_name == 'push' }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Fetch Tested Charm
+        uses: actions/download-artifact@v3
+        with:
+          name: tested-charm
+      - name: Upload charm to Charmhub edge channel
```
Comment (saltiyazan) [inline .github/workflows/main.yml:73]: why wouldn't we use `canonical/charming-actions/channel@2.0.0-rc` to select the channel as it was in `publish.yaml`?

File: .github/workflows/main.yml (hunk 1 above)
Comment (ghislainbourgeois) [reply .github/workflows/main.yml:76]: No, there is an `if` for the whole job limiting this to `push`, and the whole workflow is limited to the main branch (meaning pushes to the main branch, or PRs targeting the main branch)

File: .github/workflows/main.yml (hunk 2 above)
Comment (ghislainbourgeois) [reply .github/workflows/main.yml:73]: Good point, I can probably work that back in. The goal is to improve the `charming-actions` afterwards once the mechanics of this workflow are well tested.

Comment (ghislainbourgeois) [reply .github/workflows/main.yml:76]: Fixed

**Pull request merged**